benchmark.py

```
@@ -0,0 +1,833 @@
+    for _ in range(num_docs):
+        try:
+            fingerprint = await fingerprint_generator.__anext__()
+        except StopAsyncIteration:
+            break
+        batch.append(fingerprint)
+        
+        # Determine batch size based on mode
+        if batch_size_range[0] == batch_size_range[1]:
+            target_batch_size = batch_size_range[0]
+        else:
+            target_batch_size = random.randint(batch_size_range[0], batch_size_range[1])
```

Ruff (0.12.2)

Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)